src/vaadin-contextmenu-items-mixin.html

```diff
@@ -0,0 +1,241 @@
+      return {
+        /**
+          * @typedef MenuItem
+          * @type {object}
+          * @property {string} text - Text to be set as the menu item component's textContent
+          * @property {union: string | object} component - The component to represent the item.
+          * Either a tagName or an element instance. Defaults to "vaadin-item".
+          * @property {boolean} disabled - If true, the item is disabled and cannot be selected
+          * @property {boolean} checked - If true, the item shows a checkmark next to it
+          * @property {MenuItem[]} children - Array of child menu items
+          */
+
+        /**
+          * Defines a (hierarchical) menu structure for the component.
+          * If a menu item has a non-empty `children` set, a sub-menu with the child items is opened
+          * next to the parent menu on mouseover, tap or a right arrow keypress.
+          *
+          * The items API can't be used together with a renderer or a template!
+          *
+          * #### Example
+          *
+          * ```javascript
+          * contextMenu.items = [
+          *   {text: 'Menu Item 1', children:
+          *     [
+          *       {text: 'Menu Item 1-1', checked: true},
+          *       {text: 'Menu Item 1-2'}
+          *     ]
+          *   },
+          *   {component: 'hr'},
+          *   {text: 'Menu Item 2', children:
+          *     [
+          *       {text: 'Menu Item 2-1'},
+          *       {text: 'Menu Item 2-2', disabled: true}
+          *     ]
+          *   },
+          *   {text: 'Menu Item 3', disabled: true}
+          * ];
+          * ```
+          *
+          * @type {MenuItem[]}
+          */
+        items: Array
+      };
+    }
+
+    ready() {
+      super.ready();
+
+      // Overlay's outside click listener doesn't work with modeless
+      // overlays (submenus) so we need additional logic for it
+      this.__itemsOutsideClickListener = e => {
+        if (!e.composedPath().filter(el => el.localName === 'vaadin-context-menu-overlay')[0]) {
+          this.dispatchEvent(new CustomEvent('items-outside-click'));
+        }
+      };
+      this.addEventListener('items-outside-click', e => this.items && this.close());
+    }
+
+    connectedCallback() {
+      super.connectedCallback();
+      document.addEventListener('click', this.__itemsOutsideClickListener);
+    }
+
+    disconnectedCallback() {
+      super.disconnectedCallback();
+      document.removeEventListener('click', this.__itemsOutsideClickListener);
+    }
+
+    __openSubMenu(subMenu, itemElement) {
+      subMenu.items = itemElement._item.children;
+      subMenu.listenOn = itemElement;
+
+      const itemRect = itemElement.getBoundingClientRect();
+
+      const content = subMenu.$.overlay.$.content;
+      const padding = getComputedStyle(content).paddingTop;
+      const y = itemRect.top - parseFloat(padding);
+
+      let x;
+      subMenu.$.overlay.$.content.style.minWidth = '';
```

Comment:
BTW you can use `content.style.minWidth` here and below, as you have already stored the reference to it.

Reply:
Done.